src/routes/provider-edit-route/provider-edit-route.test.js

```diff
@@ -0,0 +1,269 @@
+              name: 'Some Proxy',
+            },
+          },
+          'proxy-id-2': {
+            id: 'proxy-id-2',
+            attributes: {
+              name: 'Some Other Proxy',
+            },
+          },
+        },
+      },
+    },
+  },
+};
+
+const rootProxy = {
+  data: {
+    attributes: {
+      proxyTypeId: 'ezproxy',
+    },
+  },
+  request: {
+    isResolved: true,
+  },
+};
+
+const mockGetProvider = jest.fn();
+const mockGetProxyTypes = jest.fn();
+const mockGetRootProxy = jest.fn();
+const mockRemoveUpdateRequests = jest.fn();
+const mockUpdateProvider = jest.fn();
+
+const getProviderEditRoute = (props = {}) => (
+  <MemoryRouter>
+    <Harness>
+      <ProviderEditRoute
+        getProvider={mockGetProvider}
+        getProxyTypes={mockGetProxyTypes}
+        getRootProxy={mockGetRootProxy}
+        history={history}
+        location={location}
+        match={match}
+        model={model}
+        proxyTypes={proxyTypes}
+        removeUpdateRequests={mockRemoveUpdateRequests}
+        rootProxy={rootProxy}
+        updateProvider={mockUpdateProvider}
+        {...props}
+      />
+        Page content
+    </Harness>
+  </MemoryRouter>
+);
+
+const renderProviderEditRoute = (props) => render(getProviderEditRoute(props));
+describe('Given ProviderEditRoute', () => {
+  beforeEach(() => {
+    mockGetProxyTypes.mockClear();
+    mockGetRootProxy.mockClear();
+    mockRemoveUpdateRequests.mockClear();
+    mockUpdateProvider.mockClear();
+    history.replace.mockClear();
+  });
+
+  afterEach(cleanup);
+
+  it('should render page', () => {
+    const { getByText } = renderProviderEditRoute({});
+
+    expect(getByText('Page content')).toBeDefined();
+  });
+
+  it('should handle getProvider with providerId', async () => {
+    await act(async () => {
+      await renderProviderEditRoute();
+    });
+
+    expect(mockGetProvider).toHaveBeenCalledWith('provider-id');
+  });
+
+  it('should handle getProvider', async () => {
+    await act(async () => {
+      await renderProviderEditRoute({
+        props: {
+          getProvider: mockGetProvider,
+          match: {
+            ...match,
+            params: { providerId: 'other-provider-id' },
+          },
+        },
+      });
+    });
+
+    expect(mockGetProvider).toHaveBeenCalled();
+  });
+
+  it('should handle GetProxyTypes', async () => {
+    await act(async () => {
+      await renderProviderEditRoute({
+        props: { proxyType : mockGetProxyTypes },
+      });
+    });
+
+    expect(mockGetProxyTypes).toHaveBeenCalled();
+  });
+
+  it('should handle GetProxyTypes', async () => {
+    await act(async () => {
+      await renderProviderEditRoute({
+        props: { proxyType : mockGetProxyTypes },
+      });
+    });
+
+    expect(mockGetProxyTypes).toHaveBeenCalled();
+  });
+
+  it('should handle Cancel', async () => {
+    let getByRole;
+
+    await act(async () => {
+      getByRole = await renderProviderEditRoute({}).getByRole;
```

Comment:
No need to pass `{}` here because you already have a default value for this argument